src/components/scenes/TransactionListScene2.tsx

```diff
@@ -0,0 +1,266 @@
+    if (typeof item === 'string') {
+      return (
+        <EdgeAnim disableAnimation={disableAnimation} enter={{ type: 'fadeInDown', distance: 30 * (index + 1) }}>
+          <SectionHeader title={item} />
+        </EdgeAnim>
+      )
+    }
+    return (
+      <EdgeAnim disableAnimation={disableAnimation} enter={{ type: 'fadeInDown', distance: 30 * (index + 1) }}>
+        <TransactionListRow navigation={navigation as NavigationBase} transaction={item} wallet={wallet} />
+      </EdgeAnim>
+    )
+  })
+
+  const keyExtractor = useHandler((item: ListItem) => {
+    if (item == null) return 'spinner'
+    if (typeof item === 'string') return item
+    return item.txid
+  })
+
+  const renderFooter: FooterRender = React.useCallback(
+    sceneWrapperInfo => {
+      return (
+        <SearchFooter
+          name="TransactionListScene2-SearchFooter"
+          placeholder={lstrings.transaction_list_search}
+          isSearching={isSearching}
+          searchText={searchText}
+          noBackground
+          sceneWrapperInfo={sceneWrapperInfo}
+          onStartSearching={handleStartSearching}
+          onDoneSearching={handleDoneSearching}
+          onChangeText={handleChangeText}
+          onLayoutHeight={handleFooterLayoutHeight}
+        />
+      )
+    },
+    [handleChangeText, handleDoneSearching, handleFooterLayoutHeight, handleStartSearching, isSearching, searchText]
+  )
+
+  return (
+    <SceneWrapper avoidKeyboard footerHeight={footerHeight} hasTabs hasNotifications renderFooter={renderFooter}>
+      {({ insetStyle, undoInsetStyle }) => (
+        <View style={undoInsetStyle}>
+          <Animated.FlatList
+            style={styles.flatList}
+            ref={flashListRef}
+            contentContainerStyle={{
+              paddingTop: insetStyle.paddingTop + theme.rem(0.5),
+              paddingBottom: insetStyle.paddingBottom + theme.rem(0.5),
+              paddingLeft: insetStyle.paddingLeft + theme.rem(0.5),
+              paddingRight: insetStyle.paddingRight + theme.rem(0.5)
+            }}
+            data={listItems}
+            keyboardDismissMode="on-drag"
+            keyboardShouldPersistTaps="handled"
+            keyExtractor={keyExtractor}
+            ListEmptyComponent={emptyComponent}
+            onEndReachedThreshold={0.5}
+            renderItem={renderItem}
+            onEndReached={handleScrollEnd}
+            onScroll={handleScroll}
+            scrollIndicatorInsets={SCROLL_INDICATOR_INSET_FIX}
+            refreshControl={refreshControl}
+          />
+        </View>
+      )}
+    </SceneWrapper>
+  )
+}
+
+/**
+ * If the token gets deleted, the scene will crash.
+ * Fall back to the main currency code if this happens.
+ */
+function checkToken(tokenId: EdgeTokenId, allTokens: EdgeTokenMap): EdgeTokenId {
+  if (tokenId == null) return null
+  if (allTokens[tokenId] == null) return null
+  return tokenId
+}
+
+export const TransactionList = withWallet(TransactionListComponent)
```

Comment:
Why weren't these components named properly? All scene components should suffix with "Scene". Also, no need for "2" suffix if we're renaming the original component.